coreneuron/nrniv/main1.cpp

```diff
@@ -87,103 +86,111 @@ void nrn_init_and_load_data(int argc, char** argv, cn_input_params& input_params
     // create mutex for nrn123, protect instance_count_
     nrnran123_mutconstruct();
 
-    // read command line parameters
-    input_params.read_cb_opts(argc, argv);
+    // read command line parameters and parameter config files
+    nrnopt_parse(argc, (const char**)argv);
 
     // set global variables
-    celsius = input_params.celsius;
-    t = input_params.celsius;
+    // precedence is: set by user, globals.dat, 34.0
+    celsius = nrnopt_get_dbl("--celsius");
+    t = celsius; // later will read globals.dat and compare with this.
 
 #if _OPENACC
-    if (!input_params.compute_gpu && input_params.cell_interleave_permute == 2) {
+    if (!nrnopt_get_flag("--gpu") && nrnopt_get_int("--cell-permute") == 2) {
         fprintf(
             stderr,
-            "compiled with _OPENACC does not allow the combination of --cell_permute=2 and missing --gpu\n");
+            "compiled with _OPENACC does not allow the combination of --cell-permute=2 and missing --gpu\n");
         exit(1);
     }
 #endif
 
     // if multi-threading enabled, make sure mpi library supports it
 #if NRNMPI
-    if (input_params.threading) {
+    if (nrnopt_get_flag("--threading")) {
         nrnmpi_check_threading_support();
     }
 #endif
 
     // full path of files.dat file
-    char filesdat_buf[1024];
-    sd_ptr filesdat = input_params.get_filesdat_path(filesdat_buf, sizeof(filesdat_buf));
+    std::string filesdat(nrnopt_get_str("--datpath") + "/" + nrnopt_get_str("--filesdat"));
 
     // reads mechanism information from bbcore_mech.dat
-    mk_mech(input_params.datpath);
+    mk_mech(nrnopt_get_str("--datpath").c_str());
 
     // read the global variable names and set their values from globals.dat
-    set_globals(input_params.datpath);
+    set_globals(nrnopt_get_str("--datpath").c_str());
 
     report_mem_usage("After mk_mech");
 
     // set global variables for start time, timestep and temperature
-    t = input_params.tstart;
+    t = nrnopt_get_dbl("--tstart");
 
-    if (input_params.dt != -1000.) {  // command line arg highest precedence
-        dt = input_params.dt;
-    } else if (dt == -1000.) {  // not on command line and no celsius in globals.dat
+    if (nrnopt_get_dbl("--dt") != -1000.) {  // command line arg highest precedence
```

Comment:
dt and celsius are normally specified by the globals.dat file. If the user changes away from
the starting sentinal values of -1000. then those supersede the globals.dat values.